src/runtime_src/core/common/api/hw_queue.cpp

```diff
@@ -267,8 +267,15 @@ class kds_device
   // execution. The execution monitor is by-passed and will be unaware
   // of argument command having been scheduled for execution.
   void
-  exec_buf(xrt_core::command* cmd)
-  {
+  exec_buf(xrt_core::command* cmd, const xrt::hw_context& hwctx)
+  {    
+    try {
+      device->exec_buf_ctx(cmd->get_exec_bo(), hwctx);
+      return;
+    }
+    catch (const xrt_core::error&) {
+        //do nothing
+    }
     device->exec_buf(cmd->get_exec_bo());
```

Comment:
No, this function should just call exec_buf_ctx, not within a try/catch.  ishim should default to call legacy exec_buf without the context.

Reply:
addressed your comment.